tests/FunctionalTestCase.php

```diff
@@ -114,6 +115,74 @@ protected function getServerStorageEngine(ReadPreference $readPreference = null)
         throw new UnexpectedValueException('Could not determine server storage engine');
     }
 
+    protected function isShardedClusterUsingReplicasets()
+    {
+        $cursor = $this->getPrimaryServer()->executeQuery(
+            'config.shards',
+            new Query([], ['limit' => 1])
+        );
+
+        $cursor->setTypeMap(['root' => 'array', 'document' => 'array']);
+        $document = current($cursor->toArray());
+
+        if (! $document ) {
+            return false;
+        }
+
+        return preg_match('@^.*/.*:\d+@', $document['host']);
```

Comment:
Is a `preg_match` actually necessary? This looks to just be testing that the URI starts with any scheme followed by a forward slash and then anything before a `:\d+` (port specifier). What distinguishes this as a replica set?

Reply:
It's necessary. With a replicaset: ``"host" : "dec6d8a7-9bc1-4c0e-960c-615f860b956f/localhost:4400,localhost:4401"``
Without a replicaset: ``"host" : "localhost:4100"``

I'll add a comment.